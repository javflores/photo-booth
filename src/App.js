import React from 'react';
import { Switch, Route } from 'react-router-dom';
import PhotoBooth from './photo-booth/index';
import PhotoStore from './store/index';
import './App.css';

const store = new PhotoStore();

const App = () => (
  <div className="App">
    <Switch>
        <Route exact path="/photobooth"
               render={props =>
                   <PhotoBooth store={store} {...props} />
               } />
    </Switch>
  </div>
);

export default App;
