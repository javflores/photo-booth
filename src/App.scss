.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

header.masthead {
  height: 90%;
  min-height: 220px;
}

.thumbnail.selected {
  border-color: black;
  border-width: 5px;
}
.thumbnail {
  cursor: pointer;
}

