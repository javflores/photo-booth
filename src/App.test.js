import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders correctly', () => {
    //In order to update the snapshop: jest --updateSnapshot

    const app = renderer.create(
        <App/>).toJSON();
    expect(app).toMatchSnapshot();
});