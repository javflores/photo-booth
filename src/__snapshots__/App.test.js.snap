// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="masthead"
    >
      <div
        className="header-content"
      >
        <div
          className="header-content-inner"
        >
          <h1
            id="homeHeading"
          >
            PhotoBooth!
          </h1>
          <hr />
        </div>
      </div>
    </header>
    <div
      className="col-lg-6 col-md-offset-3"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-3 col-xs-6"
          >
            <div
              className="selected thumbnail"
              onClick={[Function]}
            >
              <img
                alt="Baseball"
                onError={undefined}
                onLoad={undefined}
                src="http://lorempixel.com/200/200/sports/1"
              />
            </div>
          </div>
          <div
            className="col-md-3 col-xs-6"
          >
            <div
              className="thumbnail"
              onClick={[Function]}
            >
              <img
                alt="Surf"
                onError={undefined}
                onLoad={undefined}
                src="http://lorempixel.com/200/200/sports/2"
              />
            </div>
          </div>
          <div
            className="col-md-3 col-xs-6"
          >
            <div
              className="thumbnail"
              onClick={[Function]}
            >
              <img
                alt="Bike"
                onError={undefined}
                onLoad={undefined}
                src="http://lorempixel.com/200/200/sports/3"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6 col-md-offset-3"
    >
      <button
        className="col-lg-2 btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Previous
      </button>
      <button
        className="col-lg-2 btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
      <button
        className="col-lg-2 btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Random
      </button>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <h2>
          Baseball
        </h2>
      </div>
      <img
        className="row"
        src="http://lorempixel.com/500/500/sports/1"
        title="Baseball"
      />
    </div>
  </div>
</div>
`;
